api/lib/validate.js

```diff
@@ -0,0 +1,59 @@
+const _ = require('lodash');
+const errors = require('./errors.js');
+const validator = require('./schema-validator');
+const ripple = require('ripple-lib');
+
+function addOptionalOption(validator) {
```

Comment:
This is a tricky one to name. It is adding an option that represents whether the value is optional.